docs/source/deployment-experience.md

```diff
@@ -0,0 +1,135 @@
+# Deployment Experience
+
+Several facilities deploy Bluesky across many instruments (up to 25 or so).
+In this document we summarize lessons learned from that experience. At the
+moment, only NSLS-II (Bluesky's facility of origin) is represented, but we
+hope to add comments from other facilities in the future.
+
+## NSLS-II
```

Comment:
APS is a little different.  We have a central NFS share for a common set of Bluesky packages, allowing any user to use the software right away.  (We have other Python installations provided this same way.)  The shared installation is updated in the APS downtime between operating cycles (3x/year).

For production use, we recommend a custom conda environment for each instrument so they have full control of the software suite.  The environment is saved in version control with the other configuration (devices, plans, utils, ...).  We recommend GitHub.  If there are privacy concerns, then APS has an internal GitLab server to be used.

Lately, `conda` is taking very long times to solve the package requirements.  We will start to use `micromamba` to create environments and see how that is received.

We tend to avoid relying on PYTHONPATH since many beam lines use Python in other ways.  PYTHONPATH has proven to generate conflicts.  Instead, we rely on activating an environment to get all the paths correct.  For custom user libraries, we show users how to use `sys.path.append()`.